### ISSUE TITLE
"[EA] Remove unnecessary casting" inspection should handle PHP 8.0 Null-safe operator

### ISSUE THREAD

**Issue by steevechristen** [kalessil/phpinspectionsea#1861]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) |
| Language level | PHP 8.1 |

Current behaviour
--
Following code suggests "(bool) casting is now marked as unnecessary by the inspection.", which shouldn't be, since the chain can return null and therefore needs a casting (getCurrentRequest return ?Request).

```php
private function isFoo(): bool
{
  return (bool)$this->requestStack->getCurrentRequest()?->query->has('foo');
}
```
<img width="706" alt="Screenshot 2022-11-29 at 08 57 27" src="https://user-images.githubusercontent.com/15961805/204525217-95c46ff8-9a3e-4492-b799-45ccbb779cbb.png">

Expected behaviour
--
Should not mark the casting as unnecessary.

Environment details
---

-
